include/cuco/static_set.cuh

```diff
@@ -195,20 +192,38 @@ class static_set {
    */
   key_type empty_key_sentinel() const noexcept { return empty_key_sentinel_; }
 
+  // TODO rename to ref()
   /**
    * @brief Get device reference.
    *
+   * @tparam Functions Set of `cuco::functions` to be provided by the reference
+   *
    * @return Device reference of the current static_set
    */
+  template <typename... Functions>
   auto reference() const noexcept;
 
+  /**
+   * @brief Get device reference (alias definition).
+   *
+   * @tparam Functions Set of `cuco::functions` to be provided by the reference
+   *
+   * @return Device reference of the current static_set
+   */
+  template <typename... Functions>
+  auto reference_with_functions() const noexcept
```

Comment:
Bikeshed: I'd opt to just call this `reference_with` or `ref_with`. I think it will read more concisely and still be intuitive:

```
auto ref = set.ref_with<insert, find>();
```